src/v2/guide/composition.md

```diff
@@ -0,0 +1,546 @@
+}
+```
+
+And here's mixin definition that actually makes that work:
+
+``` js
+var closeMixin = {
+  // Register a close button for templates
+  components: {
+    CloseButton: {
+      template: `
+        <button v-on:click="$parent.isClosed = true">
+          X
+        </button>
+      `
+    }
+  },
+  // Track of whether the component is closed
+  data: function () {
+    return {
+      isClosed: false
+    }
+  },
+  // Simulate a v-if on the root element
+  beforeCreate: function () {
+    // Save the original, compiled render function
+    var origRender = this._render
+    // Replace the original render function
+    this._render = function _render () {
+      // If the element is closed...
+      return this.isClosed
+        // Render an empty text node
+        ? this._e()
+        // Otherwise call the original render
+        : origRender.apply(this, arguments)
+    }
+  }
+}
+```
+
+Looking at the mixin above, you might be thinking, "Wow, that looks complex!" You're right! Advanced component composition often requires deeper knowledge of both JavaScript and Vue's internals. To write the above mixin, you'd need to be familiar with:
+
+- JavaScript's [`Function.prototype.apply`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply).
+
+- Vue's internal `_render` method, which will either reference a user-provided render function or one created from a compiled, user-provided template.
+
+- Vue's internal `_e` method, which is used to create a placeholder node for falsy `v-if`s without a `v-else`. Internal helpers like this are not currently documented, but you can discover them by trying out the template you want to reproduce with Vue's [template compiler](https://vuejs.org/v2/guide/render-function.html#Template-Compilation).
+
+<p class="tip">Note that <code>_</code>-prefixed methods are not considered part of the stable API, so they may be changed or removed without warning. Using render helpers is considered relatively safe, but accompanying <a href="unit-testing.html">unit tests</a> are recommended to ensure that everything still works after upgrading Vue to a new version.</p>
+
+### Global Mixins
+
+You can also apply a mixin globally, but do this with caution!
+
+<p class="tip">Global mixins affect <strong>every</strong> Vue instance and component, including third-party components.</p>
+
+One appropriate use case, however, is to inject processing logic for custom options, such as in the [VueFire plugin](https://github.com/vuejs/vuefire#usage):
+
+``` js
+// Inject handler for a `firebase` custom option
+Vue.mixin({
+  created: function () {
+    var firebaseBindings = this.$options.firebase
+    if (firebaseBindings) {
+      // Add instance properties and listeners
+      // for data from firebase.
+    }
+  }
+})
+```
+
+Check out [VueFire's source](https://github.com/search?utf8=%E2%9C%93&q=%22Vue.mixin%22+repo%3Avuejs%2Fvuefire+path%3A%2Fsrc&type=Code) for a complete example.
+
```

Comment:
Do you think adequate context was provided to make you comfortable skipping it, or is it confusing in a way that makes you unsure if you need to understand it?

Reply:
I haven't yet used VueFire or other plugins, so when I saw that I decided it was not relevant to me right now.

Reply:
Do you think a small example with console.logs (all of a sudden every component is console logging) would be sufficient to clear it up, Katie? 

I only ask if we should add more to this to make it more clear because global mixins verge on anti-pattern

Reply:
@sdras Yeah, I do think that a `console.log` example would make it easier for me to understand what this is and see the danger of using global mixins. Good idea!

Reply:
I think `console.log` clears it up for sure. As addition though, I think it's fine to reference another project where maybe Global Mixins are used in a appropriate way. Maybe not VueFire though as the documentation there is not ideal, and it also requires people to understand Firebase to fully grock what's going on. (I recently failed with VueFire and switched to VueFirestore)